packages/core/src/extensions/Blocks/api/block.ts

```diff
@@ -70,57 +70,94 @@ export function parse<
   ];
 }
 
-// Function that uses the 'render' function of a blockConfig to create a
-// TipTap node's `renderHTML` property. Since custom blocks use node views,
-// this is only used for serializing content to the clipboard.
-export function render<
+// Function that wraps the `dom` element returned from 'blockConfig.render' in a
+// `blockContent` div, which contains the block type and props as HTML
+// attributes. If `blockConfig.render` also returns a `contentDOM`, it also adds
+// an `inlineContent` class to it.
+export function renderWithBlockStructure<
   BType extends string,
   PSchema extends PropSchema,
   ContainsInlineContent extends boolean,
-  BSchema extends BlockSchema
+  BSchema extends BlockSchema & { [k in BType]: BlockSpec<BType, PSchema> }
 >(
-  blockConfig: Omit<
-    BlockConfig<BType, PSchema, ContainsInlineContent, BSchema>,
-    "render"
-  >,
-  HTMLAttributes: Record<string, any>
+  render: BlockConfig<BType, PSchema, ContainsInlineContent, BSchema>["render"],
+  block: SpecificBlock<BSchema, BType>,
+  editor: BlockNoteEditor<BSchema>
 ) {
   // Create blockContent element
   const blockContent = document.createElement("div");
+  // Sets blockContent class
+  blockContent.className = styles.blockContent;
   // Add blockContent HTML attribute
-  blockContent.setAttribute("data-content-type", blockConfig.type);
+  blockContent.setAttribute("data-content-type", block.type);
   // Add props as HTML attributes in kebab-case with "data-" prefix
-  for (const [attribute, value] of Object.entries(HTMLAttributes)) {
-    blockContent.setAttribute(attribute, value);
+  for (const [prop, value] of Object.entries(block.props)) {
+    blockContent.setAttribute(camelToDataKebab(prop), value);
   }
 
-  // TODO: This only works for content copied within BlockNote.
-  // Creates contentDOM element to serialize inline content into.
-  let contentDOM: HTMLDivElement | undefined;
-  if (blockConfig.containsInlineContent) {
-    contentDOM = document.createElement("div");
-    blockContent.appendChild(contentDOM);
-  } else {
-    contentDOM = undefined;
+  // Renders elements
+  const rendered = render(block, editor);
+  // Add inlineContent class to inline content
+  if ("contentDOM" in rendered) {
+    rendered.contentDOM.className = `${
+      rendered.contentDOM.className
+        ? rendered.contentDOM.className + " " + styles.inlineContent
+        : styles.inlineContent
+    }`;
   }
+  // Adds elements to blockContent
+  blockContent.appendChild(rendered.dom);
 
-  return contentDOM !== undefined
+  return "contentDOM" in rendered
     ? {
         dom: blockContent,
-        contentDOM: contentDOM,
+        contentDOM: rendered.contentDOM,
       }
     : {
         dom: blockContent,
       };
 }
 
+export function getBlockFromPos<
```

Comment:
don't we have this helper function somewhere else?

Reply:
We have `getBlockInfoFromPos` but it's not that useful here since it:
- Deals more with the PM API, so it gives you nodes and positions which you'd still have to convert to a block.
- Is mostly useful since it will find the block that the position is in regardless of the positions depth and exact location inside the block. We don't really need that here since `getPos` always returns the position just before the `blockContent` node in the block.